src/main/java/com/splunk/logging/HttpEventCollectorThrowableInfo.java

```diff
@@ -0,0 +1,145 @@
+package com.splunk.logging;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Sanitised information about a throwable that is suitable for logging. Has been decoupled from
+ * any class information in the original throwable.
+ */
+public class HttpEventCollectorThrowableInfo {
+
+    private final String message;
+
+    private final String className;
+
+    private final HttpEventCollectorThrowableInfo cause;
+
+    private final List<String> stackTraceElements;
+
+    public HttpEventCollectorThrowableInfo(
+            final String message,
+            final String className,
+            final HttpEventCollectorThrowableInfo cause,
+            final List<String> stackTraceElements
+    ) {
+        super();
+        this.message = message;
+        this.className = className;
+        this.cause = cause;
+        this.stackTraceElements = stackTraceElements;
+    }
+
+    public String getClassName() { return className; }
+
+    public String getMessage() { return message; }
+
+    public HttpEventCollectorThrowableInfo getCause() { return cause; }
+
+    public List<String> getStackTraceElements() { return stackTraceElements; }
+
+    /**
+     * Builds a throwable info that includes the chain of causation with common frames removed.
+     *
+     * @param throwable the throwable from which the info will be constructed.
+     *
+     * @return the info that contains the info about the throwable.
+     */
+    public static HttpEventCollectorThrowableInfo buildFromThrowable(Throwable throwable) {
+
+        if (throwable == null) {
+            return null;
+        }
+
+        final List<Throwable> throwableList = getThrowables(throwable);
+
+        HttpEventCollectorThrowableInfo cause = null;
+        for (int i = throwableList.size() - 1; i >= 0; i--) {

```

Comment:
Need to start at the end to tie the causation chain together and preserve finality.